### ISSUE TITLE
Add support for templates

### ISSUE THREAD

**Issue by iMichka** [CastXML/CastXML#20]
Hi

Some pygccxml tests are still failing. A bunch of them expect support for templates. For example
the following struct is not dumped to the xml file:
```template <typename T> struct A {};```

I've seen comments in castxml's code that templated code is not supported for the moment. Is this something we could set up or was is not done on purpose because something is missing ?

**Comment by bradking** [comment]
Strange, gccxml never supported uninstantiated templates at all.  What does pygccxml expect that gccxml provides and CastXML doesn't?

**Comment by iMichka** [comment]
Ok, sorry I was confused and did not post everything.
In fact the real code which being tested by pygccxml is the following:

```
template <typename T> struct A {};
template <int N> struct A<const char[N]>
{ static int size(const char[N]) { return N - 1; } };
```

The resulting xml file with castxml:
https://gist.github.com/iMichka/e1faf2d67076260539c0

The resulting xml file with gccxml:
https://gist.github.com/iMichka/af55f8e8a2c56d8ed1fa

With gccxml I got:
```
<Struct id="_140" name="A&lt;const char [N]&gt;" context="_1" mangled="1AIAT__KcE" location="f1:4" file="f1" line="4" artificial="1" size="0" align="8" members="_160 " bases=""/>
```

Pyggcxml is expecting the to find the struct named in the global namespace: ```'A<const char [N]>'```

**Comment by bradking** [comment]
This was a case where gccxml would incorrectly output a *partial* template specialization as if it were a full specialization.  This was a bug that we never fixed in gccxml.  CastXML does not have the bug.

So, pygccxml should be taught to accept the lack of this output as correct or have the test case simply removed.

**Comment by iMichka** [comment]
Thanks. I found the discussion on the mailing list.
http://public.kitware.com/pipermail/gccxml/2010-January.txt

I guess the tests were just added to make sure if the bug was fixed one day in gccxml, it would detect it and trigger an error; telling the maintainer to look what has changed. It was confusing as there were no comments and I was not sure if this was the wanted output.

I kept the test for the moment and modified it for CastXML, so that it makes sures this output can not be found:
https://github.com/gccxml/pygccxml/commit/a2b12308425c6f784c379ad50194d33a2bc5aa86

Thanks for the help
